src/applications/validate-mhv-account/containers/ValidateMHVAccount.jsx

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import LoadingIndicator from '@department-of-veterans-affairs/formation-react/LoadingIndicator';
+
+import {
+  createAndUpgradeMHVAccount,
+  fetchMHVAccount,
+} from '../../../platform/user/profile/actions';
+
+import { isLoggedIn, selectProfile } from '../../../platform/user/selectors';
+import environment from '../../../platform/utilities/environment/index';
+import { replaceWithStagingDomain } from '../../../platform/utilities/environment/stagingDomains';
+
+class ValidateMHVAccount extends React.Component {
+  componentDidUpdate(prevProps) {
+    const { profile, mhvAccount } = this.props;
+    if (prevProps.profile.loading && !profile.loading) {
+      if (this.props.isLoggedIn) {
+        this.props.fetchMHVAccount();
+      } else {
+        window.location = '/';
+      }
+    }
+
+    if (prevProps.mhvAccount.loading && !mhvAccount.loading) {
+      this.redirect();
+    }
+  }
+
+  redirect = () => {
+    const { profile, mhvAccount, accountState } = this.props;
+
+    // LOA Checks
+    if (!profile.verified) {
+      window.location = 'verify';
```

Comment:
I think we should possibly be using [`window.location.replace`](https://developer.mozilla.org/en-US/docs/Web/API/Location/replace) here, so that the back button doesn't act funny.